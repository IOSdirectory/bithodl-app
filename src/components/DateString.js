import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';
import StyledText from './StyledText';

export default class DateString extends Component {
  _getDateString(date) {
    return moment(date).format('DD MMM, YYYY HH:mm');
  }

  render() {
    const dateString = this._getDateString(this.props.date);

    return (
      <StyledText style={this.props.style}>
        {dateString}
      </StyledText>
    );
  }
}

DateString.propTypes = {
  style: PropTypes.any,
  date: PropTypes.object
};
