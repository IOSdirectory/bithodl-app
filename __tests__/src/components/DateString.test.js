import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import DateString from '../../../src/components/DateString';

describe('DateString', () => {
  it('renders the date correctly', () => {
    const date = new Date('29 November, 2017, 21:59 GMT+02:00');

    const tree = renderer.create(
      <DateString date={date} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
