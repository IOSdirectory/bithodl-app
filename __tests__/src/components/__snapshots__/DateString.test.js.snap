// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateString renders the date correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={undefined}
  style={
    Array [
      Object {
        "color": "#504F63",
        "fontFamily": "Avenir Next",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  29 Nov, 2017 20:59
</Text>
`;
